api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -447,15 +438,77 @@ public static NestedField of(int id, boolean isOptional, String name, Type type,
       return new NestedField(isOptional, id, name, type, doc, null, null);
     }
 
-    public static NestedField of(
-        int id,
-        boolean isOptional,
-        String name,
-        Type type,
-        String doc,
-        Object initialDefault,
-        Object writeDefault) {
-      return new NestedField(isOptional, id, name, type, doc, initialDefault, writeDefault);
+    public static Builder from(NestedField field) {
+      return new Builder(field);
+    }
+
+    public static Builder required(String name) {
+      return new Builder(false, name);
+    }
+
+    public static Builder optional(String name) {
+      return new Builder(true, name);
+    }
+
+    public static class Builder {
```

Comment:
@wmoustafa, I'm interested to get your feedback on the builder vs new factory methods.

Reply:
Thanks Ryan. Left comment in the main conversation.